### PR TITLE
Add Triplet loss function

### PR DESCRIPTION
**Pull request by tanmayb123**

### CODE REVIEW

**Comment by tanmayb123** [comment]
One question: what should I call this? Is `triplet` just fine? Maybe `tripletLoss`?

**Comment by eaplatanios** [comment]
I feel it's nice if we can add citations to the papers that introduced the various loss functions, layers, etc., whenever they're not one of the super standard/common ones.

**Comment by eaplatanios** [comment]
@tanmayb123 Could you add a citation to [this paper](https://arxiv.org/abs/1503.03832) and also add the functional form of the loss in the documentation string?

**Comment by tanmayb123** [comment]
Sure thing, but @rxwei had decided some time ago _not_ to include loss functions that aren't available in Keras, but this PR stayed open. What should I do?

**Comment by rxwei** [comment]
Let's add it for now!

File: Sources/DeepLearning/Loss.swift [hunk 1]
```diff
@@ -53,3 +53,21 @@ public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
         (Tensor<Scalar>(1) - labels) * log(Tensor<Scalar>(1) - logits)
     return -loss.mean(alongAxes: 0).sum()
 }
+
+/// Computes the triplet loss.
```
Comment (rxwei) [inline Sources/DeepLearning/Loss.swift]:
```suggestion
/// Returns the triplet loss.
```

Comment (rxwei) [reply Sources/DeepLearning/Loss.swift]: Also add the citation as Anthony suggested.

**Comment by saeta** [comment]
Hi @tanmayb123 ! It looks like this PR is coming from an unknown repository. As a result, we'll close this out for now. But if you'd like to try and address the comments and add a test, I think we'd love to get this merged in. 

All the best,
-Brennan

**Comment by Kshitij09** [comment]
@tanmayb123 should I borrow your code and submit a new PR with requested changes? I was planning to submit a PR on your fork but seems like the branch having this code missing.
